tests/test_helper/bats/bgp_data_plane.bats

```diff
@@ -0,0 +1,139 @@
+    echo "# Starting BGP in $BGP_PEER on interface $BGP_CONTAINER_INT_IFACE" >&3
+    frr_start_bgp_unnumbered "$BGP_PEER" "$BGP_CONTAINER_INT_IFACE" "$tor_asn"
+
+    # Enable BGP redirection and start BGP daemon in OVN chassis
+    local host_asn=4210000000
+    local BGP_NET_IP="172.16.10.1/24"
+    local vrf="10"
+    local vrf_device="ovnvrf$vrf"
+    local external_connections="$OVN_CONTAINER_INT_IFACE:$BGP_NET_IP"
+
+    echo "# Enabling MicroOVN BGP in $TEST_CONTAINER and configuring BGP (ASN $host_asn)" >&3
+    lxc_exec "$TEST_CONTAINER" "microovn enable bgp \
+        --config ext_connection=$external_connections \
+        --config vrf=$vrf \
+        --config asn=$host_asn"
+
+    echo "# ($TEST_CONTAINER) waiting on established BGP with $BGP_PEER" >&3
+    wait_until "microovn_bgp_established $TEST_CONTAINER $vrf_device $BGP_PEER"
+
+    # create VIF that represents VM on the internal OVN network
+    local gw_lr="lr-$TEST_CONTAINER-microovn"
+    local guest_ls="ls-guest-net"
+    local guest_lsp_to_lr="lsp-to-gw"
+    local lrp_to_guest_ls="lrp-to-guest"
+    local guest_lrp_ip="192.168.10.1"
+    local guest_lrp_cidr="$guest_lrp_ip/24"
+    local guest_vm_ip="192.168.10.10"
+    local guest_vm_cidr="$guest_vm_ip/24"
+    local guest_vm_mac="02:00:aa:00:00:01"
+    local guest_vm_lsp="lsp-guest-vm"
+    local guest_vm_iface="guest-vm"
+    local guest_vm_ns="ns-guest"
+
+    echo "# ($TEST_CONTAINER) Create VIF in the OVN network that represents Virtual Machine with IP $guest_vm_cidr" >&3
+    echo "# ($TEST_CONTAINER) Create OVN network '$guest_ls' and connect it to router '$gw_lr' ($guest_lrp_cidr)"
+    lxc_exec "$TEST_CONTAINER" \
+        "microovn.ovn-nbctl \
+         -- \
+         lrp-add $gw_lr $lrp_to_guest_ls 02:00:ff:00:00:01 $guest_lrp_cidr \
+         -- \
+         ls-add $guest_ls \
+         -- \
+         lsp-add $guest_ls $guest_lsp_to_lr \
+         -- \
+         lsp-set-type $guest_lsp_to_lr router \
+         -- \
+         lsp-set-options $guest_lsp_to_lr router-port=$lrp_to_guest_ls \
+         -- \
+         lsp-set-addresses $guest_lsp_to_lr router \
+         "
+
+    echo "# ($TEST_CONTAINER) Create LSP '$guest_vm_lsp' ($guest_vm_cidr)"
+    lxc_exec "$TEST_CONTAINER" \
+        "microovn.ovn-nbctl \
+         -- \
+         lsp-add $guest_ls $guest_vm_lsp \
+         -- \
+         lsp-set-addresses $guest_vm_lsp '$guest_vm_mac $guest_vm_cidr'"
+
+    echo "# ($TEST_CONTAINER) Create netns '$guest_vm_ns' and move VM interface '$guest_vm_iface' into it"
+    lxc_exec "$TEST_CONTAINER" "ip netns add $guest_vm_ns"
+    lxc_exec "$TEST_CONTAINER" \
+        "microovn.ovs-vsctl \
+         -- \
+         add-port br-int $guest_vm_iface \
+         -- \
+         set Interface $guest_vm_iface type=internal external_ids:iface-id=$guest_vm_lsp"
+    netns_ifadd "$TEST_CONTAINER" "$guest_vm_ns" "$guest_vm_iface" "$guest_vm_mac" "$guest_vm_cidr"
+    wait_until "microovn_lsp_up $TEST_CONTAINER $guest_vm_lsp"
```

Comment:
I found the function to be tightly coupled with the `microovn_add_gw_router` which creates LS required by this function and connects it to the LR. However seeing as nothing currently uses `microovn_add_vif`, perhaps it could be modified to take `ls_name` as an argument.

Reply:
It is in use on main, and adding an optional parameter to supply `ls_name` would be nice.